components/apimgt/org.wso2.carbon.apimgt.gateway/src/test/java/org/wso2/carbon/apimgt/gateway/handlers/security/basicauth/BasicAuthAuthenticatorTest.java

```diff
@@ -0,0 +1,148 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.apimgt.gateway.handlers.security.basicauth;
+
+
+import io.swagger.models.Swagger;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.wso2.carbon.apimgt.gateway.APIMgtGatewayConstants;
+import org.wso2.carbon.apimgt.gateway.handlers.security.APISecurityConstants;
+import org.wso2.carbon.apimgt.gateway.handlers.security.APISecurityException;
+import org.wso2.carbon.apimgt.gateway.handlers.security.AuthenticationResponse;
+
+import java.util.TreeMap;
+
+public class BasicAuthAuthenticatorTest {
+    private MessageContext messageContext;
+    private org.apache.axis2.context.MessageContext axis2MsgCntxt;
+    private BasicAuthAuthenticator basicAuthAuthenticator;
+    private final String CUSTOM_AUTH_HEADER = "AUTH-HEADER";
+
+    @Before
+    public void setup() {
+        messageContext = Mockito.mock(Axis2MessageContext.class);
+        axis2MsgCntxt = Mockito.mock(org.apache.axis2.context.MessageContext.class);
+        Mockito.when(axis2MsgCntxt.getProperty(APIMgtGatewayConstants.REQUEST_RECEIVED_TIME)).thenReturn("1506576365");
+        Mockito.when(((Axis2MessageContext) messageContext).getAxis2MessageContext()).thenReturn(axis2MsgCntxt);
+
+        basicAuthAuthenticator = new BasicAuthAuthenticator(CUSTOM_AUTH_HEADER, true, null);
+        basicAuthAuthenticator.setBasicAuthCredentialValidator(new BasicAuthCredentialValidator() {
+            @Override
+            public boolean validate(String username, String password) {
+                if ((username.equals("test_username@carbon.super") ||
+                        username.equals("test_username_blocked@carbon.super")) && password.equals("test_password")) {
+                    return true;
+                }
+                return false;
+            }
+
+            @Override
+            public boolean validateScopes(String username, Swagger swagger, MessageContext synCtx) throws APISecurityException {
+                if (username.equals("test_username@carbon.super")) {
+                    return true;
+                } else if (username.equals("test_username_blocked@carbon.super")) {
+                    throw new APISecurityException(APISecurityConstants.INVALID_SCOPE, "Scope validation failed");
+                }
+                return false;
+            }
```

Comment:
This is not required, the same behaviour can be achieved by creating a Mock BasicAuthCredentialValidator and then specifying using Mockito.when how validate() and validateScopes() will behave.

Reply:
Updated